packages/apollo-engine-reporting/src/agent.ts

```diff
@@ -421,12 +470,41 @@ class ReportData {
     });
     this.reset();
   }
+
   reset() {
     this.report = new Report({ header: this.header });
     this.size = 0;
   }
 }
 
+class StatsMap {
+  readonly map: { [k: string]: ContextualizedStats } = Object.create(null);
```

Comment:
You have at least three cases of defining "maps" with this pattern (search for `string]`). However, this tells TypeScript that literally any key you pass in will return a ContextualizedStats object, which means that TS doesn't help you find cases where you forget to handle key not found properly. I suggest you either use `Map`, or you explicitly put `ContextualizedStats | undefined` or something.

...

except, bah, in some cases (not this one, but eg TypeStat.perFieldStat) you can't do that because protobufjs's generated types are bad. honestly maybe we should just fix that?  We already are running a fork of protobufjs. The code in question is https://github.com/apollographql/protobuf.js/blob/9d11c6afd940b3ffd84731290c78a834ef2db477/cli/lib/tsd-jsdoc/publish.js#L289-L292 , we can just add `| undefined`?

https://github.com/apollographql/protobuf.js/pull/5 thoughts?